realm/realm-library/src/androidTest/java/io/realm/entities/AllJavaTypes.java

```diff
@@ -44,6 +44,8 @@
     public static final String   INVALID_LINKED_BINARY_FIELD_FOR_DISTINCT = AllJavaTypes.FIELD_OBJECT + "." + AllJavaTypes.FIELD_BINARY;
     public static final String[] INVALID_LINKED_TYPES_FIELDS_FOR_DISTINCT = new String[]{FIELD_OBJECT + "." + FIELD_BINARY, FIELD_OBJECT + "." + FIELD_OBJECT, FIELD_OBJECT + "." + FIELD_LIST};
 
+    public static final long DEFAULT_PRIMARYKEY_INDEX = 3;
```

Comment:
I would like to not assume pk index as a constant 3. Instead, it will be more helpful by checking if the PK index is correct (than checking the result decreased every time). I will decouple the test case from the `Table` implementation to make it still useful after we moved to OS implementation.